### PR TITLE
feature: add "version" label to all prometheus metrics

### PR DESCRIPTION
**Pull request by szuecs**
feature: add "version" label to all prometheus metrics

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

**Comment by MustafaSaber** [comment]
👍

File: metrics/prometheus.go [hunk 1]
```diff
@@ -54,6 +56,9 @@ type Prometheus struct {
 
 // NewPrometheus returns a new Prometheus metric backend.
 func NewPrometheus(opts Options) *Prometheus {
+	if version == "" {
+		version = "<unknown>"
+	}
```
Comment (AlexanderYastrebov) [inline metrics/prometheus.go:61]: Why not pass version in the Options and set it via already available main.version?

Comment (szuecs) [reply metrics/prometheus.go:61]: This discussion we already had.
The reason is that with how we do this now in the linker is the best option to not trick ourselves.
It's simple and works at build time.

Comment (AlexanderYastrebov) [reply metrics/prometheus.go:61]: I struggle to see how setting an obscure global variable via linker (which also must have the same value as an existing variable) is better then passing it via Options. Library users must update their Makefiles

Comment (AlexanderYastrebov) [reply metrics/prometheus.go:61]: It should be also possible to add version label from pod labels during scarping

Comment (AlexanderYastrebov) [reply metrics/prometheus.go:61]: > It should be also possible to add version label from pod labels during scarping

See https://github.com/zalando-incubator/kubernetes-on-aws/pull/6579

Comment (szuecs) [reply metrics/prometheus.go:61]: The question is what is more ugly.
IMO linker variable is not really obscure, but the labeling of an independently configured metrics store actually is. :)

**Pull request merged**